### PR TITLE
WAI-1022 tonga covid raw data download

### PR DESCRIPTION
**Pull request by EMcQ-BES**
### Changes:

- Add migration for custom data download for Tonga Covid Contact Tracing
- Bugfix issue with multiple `entityIdToName` translations

### CODE REVIEW

File: packages/web-config-server/src/apiV1/dataBuilders/helpers/translateEventEntityIdsToNames.js [hunk 1]
```diff
@@ -8,14 +8,20 @@ import { transformValue } from 'apiV1/dataBuilders/transform';
...
+          const acc = await accPromise;
+          const dataValueEntityId = event.dataValues[dataElementCode];
+          if (dataValueEntityId) {
+            const entityName = await transformValue(models, 'entityIdToName', dataValueEntityId);
+            return { ...acc, [dataElementCode]: entityName };
+          }
+          return acc;
+        },
+        {},
+      );
```
Comment (edmofro) [inline packages/web-config-server/src/apiV1/dataBuilders/helpers/translateEventEntityIdsToNames.js:24]: This is cool, and obviously works well, but honestly I think it'd be less confusing to just mutate a variable that's declared above a `.map` of promises (or even a classic `for of` loop for that matter). Javascript functional patterns are all well and good, but sometimes they bury the lede imo
```suggestion
      const updatedDataValues = {};
      await Promise.all(dataElementCodes.map(async dataElementCode => {
        const dataValueEntityId = event.dataValues[dataElementCode];
        if (dataValueEntityId) {
          const entityName = await transformValue(models, 'entityIdToName', dataValueEntityId);
          updatedDataValues[dataElementCode] = entityName;
        }
      }));
```

**Review by edmofro — CHANGES_REQUESTED** [review]
Migration looks great (and fast work!)

File: packages/web-config-server/src/apiV1/dataBuilders/helpers/translateEventEntityIdsToNames.js (hunk 1 above)
Comment (EMcQ-BES) [reply packages/web-config-server/src/apiV1/dataBuilders/helpers/translateEventEntityIdsToNames.js]: Oh that's a good call. I think when bug-fixing my mind tends to focus on finding the issue in the written code, and I don't consider improvements while I'm there.

**Review by edmofro — APPROVED** [review]
LGTM

Can you create a new branch `release-2021-44-1` (as per https://beyond-essential.slab.com/posts/release-process-j4ersmrg#h4ytu-step-2-final-merge-from-dev-at-11-am step 2.2) and merge this into it? We'll then add https://github.com/beyondessential/tupaia/pull/3451 and release them together after some regression testing tomorrow morning.

**Pull request merged**